src/Store/Azure/SearchStore.php

```diff
@@ -73,23 +69,23 @@ private function request(string $endpoint, array $payload): array
     /**
      * @return array<string, mixed>
      */
-    private function convertDocumentToIndexableArray(Document $document): array
+    private function convertDocumentToIndexableArray(VectorDocument $document): array
     {
         return array_merge([
             'id' => $document->id,
             $this->vectorFieldName => $document->vector->getData(),
+            'text' => $document->text,
         ], $document->metadata->getArrayCopy());
     }
 
     /**
      * @param array<string, mixed> $data
      */
-    private function convertArrayToDocument(array $data): Document
+    private function convertArrayToDocument(array $data): VectorDocument
     {
-        return new Document(
+        return new VectorDocument(
             id: Uuid::fromString($data['id']),
-            text: null,
-            vector: null,
+            vector: $data[$this->vectorFieldName] ? new Vector($data[$this->vectorFieldName]) : null,
```

Comment:
I would assume, that the vector is available, I would assume the data was added via our Store classes

Reply:
no, wait, it's not necessarily retrievable